### PR TITLE
feature benchmark: repeated scenario runs

### PR DESCRIPTION
**Pull request by nrainer-materialize**
This implements #29590.

### Nightly
https://buildkite.com/materialize/nightly/builds?branch=nrainer-materialize%3Afeature-benchmark%2Frepeated-scenario-runs

### CODE REVIEW

File: misc/python/materialize/feature_benchmark/benchmark_versioning.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 from materialize import MZ_ROOT
 
-FEATURE_BENCHMARK_FRAMEWORK_VERSION = "1.3.0"
+FEATURE_BENCHMARK_FRAMEWORK_VERSION = "1.4.0"
```
Comment (nrainer-materialize) [inline misc/python/materialize/feature_benchmark/benchmark_versioning.py:13]: Not sure if this change warrants upgrading the version.

**Comment by nrainer-materialize** [comment]
I still need to validate and test the changes.

File: misc/python/materialize/feature_benchmark/benchmark_versioning.py (hunk 1 above)
Comment (def-) [reply misc/python/materialize/feature_benchmark/benchmark_versioning.py:13]: In metabase can we easily merge multiple versions? If yes, then the bump is fine and we can just display them together for now.

**Review by def- — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
This causes way more flakes. We need to discuss at the onsite whether we want to proceed with this change and possibly increase thresholds or discard it.

**Comment by def-** [comment]
In my opinion we should keep feature-benchmark as is so we can keep catching regressions, and improve parallel-benchmark to be the one-run-reliable benchmarking framework. First issue about where parallel-benchmark was not entirely consistent: https://github.com/MaterializeInc/materialize/issues/29653

**Comment by def-** [comment]
I'll rebase this on top of https://github.com/MaterializeInc/materialize/pull/29664, maybe that helps?

File: misc/python/materialize/feature_benchmark/benchmark_versioning.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/feature_benchmark/benchmark_versioning.py:13]: Not really. Reverting the version upgrade

**Comment by nrainer-materialize** [comment]
> In my opinion we should keep feature-benchmark as is so we can keep catching regressions

I changed this PR to still always conduct three runs per scenario but pick the best outcome (instead of the median outcome) with https://github.com/MaterializeInc/materialize/pull/29637/commits/7478e73c66e48f9f10b3d9cae09887945f1130c0.

**Comment by def-** [comment]
Good compromise

**Pull request merged**